File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	/**
	 * 查询所有
	 * @return
	 */
	public List<TbBrand> findAll();
	
	/**
	 * 分页查询
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 添加品牌
	 * @return
	 */
	public void add(TbBrand tbBrand);
	
	/**
	 * 修改品牌
	 * @param tbBrand
	 */
	public void update(TbBrand tbBrand);
	
	/**
	 * 通过ID查询TbBrand
	 * @param tbBrand
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 通过ID数组删除多个对象
	 * @param id
	 * @return
	 */
	public void  delete(Long[] ids);
	
	/**
	   * 条件查询
	 * @param tbBrand
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(TbBrand tbBrand ,int pageNum,int pageSize);


	/**
	 * 关联品牌下拉数据
	 * @return
	 */
	List<Map> selectOptionList();

	
		
	
	
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){    //品牌服务层
				//读取列表数据绑定到表单中
				this.findAll=function(){
					return $http.get('../brand/findAll.do');
				}
				
				this.findPage=function(page,rows){
					return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
				}

				this.save=function(){
						var methodName="add";
						if($scope.entity.id!=null){
							methodName="update";
						}
					return	$http.post('../brand/'+methodName+'.do',$scope.entity);
				}

				this.findOne=function(id){
					return $http.get('../brand/findOne.do?id='+id);
				}
				
				this.dele=function(){
					return $http.get('../brand/delete.do?ids='+$scope.selectIds);
				}

				this.search=function(page,rows,searchEntity){
					return $http.post('../brand/search.do?page='+page+'&rows='+rows,searchEntity);
				}

				this.selectOptionList=function () {
					return $http.get('../brand/selectOptionList.do');
                }
				
			});